src/region.js

```diff
@@ -321,46 +247,13 @@ var Region = MarionetteObject.extend({
       // to throw a ViewDestroyedError for this view
       view._isDestroyed = true;
     }
+    if (shouldTriggerDetach) {
+      triggerMethodOn(view, 'detach', view);
+    }
     if (!view.supportsDestroyLifecycle) {
       triggerMethodOn(view, 'destroy', view);
     }
-  },
-
-  // Attach an existing view to the region. This
-  // will not call `render` or `onShow` for the new view,
-  // and will not replace the current HTML for the `el`
-  // of the region.
-  attachView: function(view) {
-    if (this.currentView) {
-      delete this.currentView._parent;
-    }
-    view._parent = this;
-    this.currentView = view;
-    return this;
-  },
-
-  // Checks whether a view is currently present within
-  // the region. Returns `true` if there is and `false` if
-  // no view is present.
-  hasView: function() {
-    return !!this.currentView;
-  },
```

Comment:
Well, I don't know of a recommended pattern that uses `attachView`, I wonder if it's more of an anti-pattern to get that low level with a Region.  I already added back `hasView` locally this morning, though--will push soon.  But basically I had a feeling I would err on the side of slightly over-aggressive API reduction, so I wanted your thoughts.  I'm fine adding it back in.